torch/_dynamo/utils.py

```diff
@@ -3067,3 +3068,42 @@ def get_user_object_from_id(obj_id):
 def store_user_object_weakref(obj):
     obj_id = id(obj)
     user_obj_id_to_weakref[obj_id] = weakref.ref(obj)
+
+
+class CompileTimeInstructionCounter:
+    _counter: int = 0
+    _id: Optional[int] = None
+
+    @classmethod
+    def start(cls):
+        if cls._id is not None:
+            raise RuntimeError("CompileTimeInstructionCounter is already started")
+        cls._id = i_counter.start()
+
+    @classmethod
+    def end(cls):
+        cls._counter += i_counter.end(cls._id)
+        cls._id = None
+
+    @classmethod
+    def clear(cls):
+        cls._counter = 0
+
+    @classmethod
+    def value(cls):
+        return cls._counter
+
+
+def maybe_count_instructions(func):
+    if config.record_compile_time_instruction_count:
+
+        @functools.wraps(func)
+        def wrapper_function(*args, **kwargs):
```

Comment:
No decorator please, it makes the cProfiles worse. Do this as a context manager.

Reply:
I can do that.